applications/DEMApplication/custom_elements/beam_particle.cpp

```diff
@@ -0,0 +1,392 @@
+        const array_1d<double, 3>& ang_vel     = this->GetGeometry()[0].FastGetSolutionStepValue(ANGULAR_VELOCITY);
+
+        int NeighbourSize = mNeighbourElements.size();
+        GetGeometry()[0].GetSolutionStepValue(NEIGHBOUR_SIZE) = NeighbourSize;
+
+        for (int i = 0; data_buffer.SetNextNeighbourOrExit(i); ++i) {
+
+            if (mNeighbourElements[i] == NULL) continue;
+            if (this->Is(NEW_ENTITY) && mNeighbourElements[i]->Is(NEW_ENTITY)) continue;
+            SphericContinuumParticle* neighbour_iterator = dynamic_cast<SphericContinuumParticle*>(mNeighbourElements[i]);
+            data_buffer.mpOtherParticle = neighbour_iterator;
+
+            unsigned int neighbour_iterator_id = data_buffer.mpOtherParticle->Id();
+
+            noalias(data_buffer.mOtherToMeVector) = this->GetGeometry()[0].Coordinates() - data_buffer.mpOtherParticle->GetGeometry()[0].Coordinates();
+
+            const double& other_radius = data_buffer.mpOtherParticle->GetRadius();
+
+            data_buffer.mDistance = DEM_MODULUS_3(data_buffer.mOtherToMeVector);
+
+            double radius_sum = GetRadius() + other_radius;
+            double initial_delta = GetInitialDelta(i);
+            double initial_dist = radius_sum - initial_delta;
+            double indentation = initial_dist - data_buffer.mDistance;
+
+            double myYoung = GetYoung();
+            double myPoisson = GetPoisson();
+
+            double kn_el = 0.0;
+            double kt_el_0 = 0.0;
+            double kt_el_1 = 0.0;
+            double DeltDisp[3] = {0.0};
+            double RelVel[3] = {0.0};
+            DEM_SET_COMPONENTS_TO_ZERO_3x3(data_buffer.mLocalCoordSystem)
+            DEM_SET_COMPONENTS_TO_ZERO_3x3(data_buffer.mOldLocalCoordSystem)
+            bool sliding = false;
+
+            double contact_tau = 0.0;
+            double contact_sigma = 0.0;
+            double failure_criterion_state = 0.0;
+            double acumulated_damage = 0.0;
+
+            // Getting neighbor properties
+            double other_young = data_buffer.mpOtherParticle->GetYoung();
+            double other_poisson = data_buffer.mpOtherParticle->GetPoisson();
+            double equiv_poisson;
+            if ((myPoisson + other_poisson) != 0.0) { equiv_poisson = 2.0 * myPoisson * other_poisson / (myPoisson + other_poisson); }
+            else { equiv_poisson = 0.0; }
+
+            double equiv_young = 2.0 * myYoung * other_young / (myYoung + other_young);
+            double calculation_area = 0.0;
+            const double equiv_shear = equiv_young / (2.0 * (1 + equiv_poisson));
+
+            if (i < (int)mContinuumInitialNeighborsSize) {
+                double area = this->GetProperties()[BEAM_CROSS_SECTION];
+                double other_area = data_buffer.mpOtherParticle->GetProperties()[BEAM_CROSS_SECTION];
+                calculation_area = std::max(area, other_area);
+                mContinuumConstitutiveLawArray[i]->CalculateElasticConstants(kn_el,
```

Comment:
Provided that you are creating a new element for this purpose, here you don't need to call a ContinuumCL. You could call a BeamCL directly. This has some benefitial consequences:
1 - You don't need to modify the base clase ContinuumCL
2-  Your element would have two arrays in memory: one with ContinuumCL's and another with BeamCL's. In this way, the beam can be stuck in a continuum by keeping bonds with continuum particles and can be part of a beam as well simultaneously. If so, the properties should have information about both, of course.

Reply:
Thanks for the explanation. I made the changes to call a BeamCL directly.